itests/hcatalog-unit/src/test/java/org/apache/hive/hcatalog/listener/TestDbNotificationCleanup.java

```diff
@@ -0,0 +1,210 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.hive.hcatalog.listener;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.fail;
+
+import java.util.concurrent.TimeUnit;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.hadoop.hive.cli.CliSessionState;
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.hadoop.hive.metastore.HiveMetaStoreClient;
+import org.apache.hadoop.hive.metastore.IMetaStoreClient;
+import org.apache.hadoop.hive.metastore.IMetaStoreClient.NotificationFilter;
+import org.apache.hadoop.hive.metastore.api.NotificationEventRequest;
+import org.apache.hadoop.hive.metastore.api.NotificationEventResponse;
+import org.apache.hadoop.hive.metastore.api.Database;
+import org.apache.hadoop.hive.metastore.conf.MetastoreConf;
+import org.apache.hadoop.hive.metastore.messaging.MessageDeserializer;
+import org.apache.hadoop.hive.metastore.messaging.json.JSONMessageEncoder;
+import org.apache.hadoop.hive.ql.DriverFactory;
+import org.apache.hadoop.hive.ql.IDriver;
+import org.apache.hadoop.hive.ql.session.SessionState;
+import org.apache.hive.hcatalog.api.repl.ReplicationV1CompatRule;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.AfterClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestRule;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import static org.apache.hadoop.hive.metastore.conf.MetastoreConf.ConfVars.EVENT_DB_LISTENER_CLEAN_STARTUP_WAIT_INTERVAL;
+
+
+
+public class TestDbNotificationCleanup {
+    private static final Logger LOG = LoggerFactory.getLogger(TestDbNotificationCleanup.class
+            .getName());
+    private static final int EVENTS_TTL = 30;
+    private static final int CLEANUP_SLEEP_TIME = 10;
+    private static Map<String, String> emptyParameters = new HashMap<String, String>();
+    private static IMetaStoreClient msClient;
+    private static IDriver driver;
+    private static MessageDeserializer md;
+    private static HiveConf conf;
+
+    private long firstEventId;
+    private final String testTempDir = Paths.get(System.getProperty("java.io.tmpdir"), "testDbNotif").toString();
+
+    private static List<String> testsToSkipForReplV1BackwardCompatTesting =
+            new ArrayList<>(Arrays.asList("cleanupNotifs", "cleanupNotificationWithError","skipCleanedUpEvents"));
+    // Make sure we skip backward-compat checking for those tests that don't generate events
+
+    private static ReplicationV1CompatRule bcompat = null;
+
+    @Rule
+    public TestRule replV1BackwardCompatibleRule = bcompat;
```

Comment:
This entire section is not required. you set all 3 tests to skip, and there is no more tests in this class. as a result the rule can be simply removed.

Reply:
Fixed